metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableOpWrapper.java

```diff
@@ -44,14 +67,31 @@
      */
     public Map<String, ScanSummary.PartitionMetrics> getPartitionMetricsMap(final Table icebergTable,
                                                                      @Nullable final Expression filter) {
-        return (filter != null)
-            ? ScanSummary.of(icebergTable.newScan().filter(filter))
-                .limit(connectorContext.getConfig().getMaxPartitionsThreshold())
-                .throwIfLimited()
-                .build()
+        Map<String, ScanSummary.PartitionMetrics> result = Maps.newHashMap();
+        //
+        // Cancel the iceberg call if it times out.
+        //
+        final Future<Map<String, ScanSummary.PartitionMetrics>> future = threadServiceManager.getExecutor()
+            .submit(() -> (filter != null) ? ScanSummary.of(icebergTable.newScan().filter(filter))
+            .limit(config.getMaxPartitionsThreshold())
+            .throwIfLimited()
+            .build()
             :
             ScanSummary.of(icebergTable.newScan())  //the top x records
-                .limit(connectorContext.getConfig().getIcebergTableSummaryFetchSize())
-                .build();
+                .limit(config.getIcebergTableSummaryFetchSize())
+                .build());
+        try {
+            final int getIcebergPartitionsTimeout = Integer.parseInt(configuration
+                .getOrDefault(HiveConfigConstants.GET_ICEBERG_PARTITIONS_TIMEOUT, "120"));
+            result = future.get(getIcebergPartitionsTimeout, TimeUnit.SECONDS);
```

Comment:
Feel 120 seconds is quite short, not sure if how many tables can survive it. Maybe 5 minutes?

Reply:
120 secs is the default. We can always configure it.